addons/mail/static/src/components/notification_list/notification_list.js

```diff
@@ -133,9 +133,8 @@ class NotificationList extends Component {
         if (props.filter === 'all') {
             const notificationGroups = this.env.messaging.notificationGroupManager.groups;
             notifications = Object.values(notificationGroups)
-                .sort((group1, group2) =>
-                    group1.date.isAfter(group2.date) ? -1 : 1
-                ).map(notificationGroup => {
+                .sort((group1, group2) => group1.sequence < group2.sequence ? -1 : 1)
```

Comment:
I think this is the problem with the test that I don't like the change.

This should be `group1.sequence > group2.sequence` instead.

Most recent will have the bigger sequence because it's based on id, and we want it first so it should return -1 for it.

Reply:
Please adapt how `sequence` values are set and keep current ordering based on `sequence`.
The term "sequence" is well-used for solving ordering and the general idea is lower sequence comes first. If we do the opposite, it quickly becomes confusing.